tests/src/functional/Helpers/EnvironmentDetectorTest.php

```diff
@@ -0,0 +1,157 @@
+    $this->composer = new Composer();
+    $package = new RootPackage("acquia/drupal-recommended-settings", "1.0", "1.0.0");
+    $package->setExtra([
+      "drupal-scaffold" => [
+        "locations" => [
+          "web-root" => "docroot"
+        ],
+      ],
+    ]);
+    $this->composer->setPackage($package);
+    $this->composer->setConfig($config);
+    $this->plugin = new Plugin();
+    $this->io = $this->createMock(IOInterface::class);
+    $activateMethod = $this->getReflectionMethod(Plugin::class, "activate");
+    $activateMethod->invokeArgs($this->plugin, [$this->composer, $this->io]);
+
+    $getDrupalRootMethod = $this->getReflectionMethod(Plugin::class, "getDrupalRoot");
+    $this->drupalRoot = $drupalRootDir = $getDrupalRootMethod->invoke($this->plugin);
+    // Define DRUPAL_ROOT once.
+    if (!defined('DRUPAL_ROOT')) {
+      define('DRUPAL_ROOT', $drupalRootDir);
+    }
+    // Generate folder/files for ACSF.
+    $drsFileSystem = new DrsFilesystem();
+    $drsFileSystem->ensureDirectoryExists($drupalRootDir . '/sites/g');
+    $drsFileSystem->dumpFile($drupalRootDir . '/sites/g/random.php', "<?php echo 'hello';");
+  }
+
+  /**
+   * Tests EnvironmentDetector::getCiEnv().
+   *
+   * @throws \ReflectionException
+   */
+  public function testGetCiEnv(): void {
+
+    putenv("PIPELINE_ENV=TRUE");
+    $this->assertSame('pipelines', EnvironmentDetector::getCiEnv());
+
+    putenv("PIPELINE_ENV=");
+    putenv("GITLAB_CI_TOKEN=TRUE");
+    $this->assertSame('codestudio', EnvironmentDetector::getCiEnv());
+
+    putenv("PIPELINE_ENV=");
+    putenv("GITLAB_CI_TOKEN=");
+    putenv("CI=TRUE");
+    $this->assertTrue(EnvironmentDetector::isCiEnv() );
+  }
+
+  /**
+   * Verify Ci settings file suggestion exists.
+   *
+   * @throws \ReflectionException
+   */
+  public function testGetCiSettingsFile(): void {
+    putenv("PIPELINE_ENV=TRUE");
+    $this->assertStringEndsWith('/acquia/drupal-recommended-settings/settings/pipelines.settings.php', EnvironmentDetector::getCiSettingsFile());
+  }
+
+  /**
+   * Verify multiple environment.
+   *
+   * @throws \ReflectionException
+   */
+  public function testMultipleEnv(): void {
+    putenv("PIPELINE_ENV=");
+    putenv("CI=");
+    $this->assertTrue(EnvironmentDetector::isLocalEnv());
+
+    putenv("AH_SITE_ENVIRONMENT=dev");
+    $this->assertTrue(EnvironmentDetector::isDevEnv());
+
+    putenv("AH_SITE_ENVIRONMENT=test");
+    $this->assertTrue(EnvironmentDetector::isStageEnv());
+
+    putenv("AH_SITE_ENVIRONMENT=prod");
+    $this->assertTrue(EnvironmentDetector::isProdEnv());
+  }
+
+  /**
+   * Test EnvironmentDetector::isAcsfInited().
+   */
+  public function testIsAcsfInited(): void {
+    $this->assertTrue(EnvironmentDetector::isAcsfInited());
+  }
+
+  /**
+   * Test EnvironmentDetector::getSiteName().
+   * @throws \ReflectionException
+   */
+  public function testGetSiteName(): void {
+    $sitePatch = "sites/default";
```

Comment:
@chandan-singh7929 The method `getSiteName()` returns site name based on different scenarios. Let's see if we can add those and validate all this scenarios.